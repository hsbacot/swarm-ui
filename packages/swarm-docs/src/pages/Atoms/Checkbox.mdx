import { Checkbox } from '@meetup/swarm-components'
import Example from '../../components/examples/checkbox'
import PropsTable from '../../components/propstable';

# Checkbox
---
A checkbox allows a user to select a value from a small set of options, often binary.

## Example
<Example />

# Snippets
```.jsx
<>
<Checkbox label="do this thing" checked={true === true} />
<Checkbox label="also this thing" checked={true === false} />
</>
```

# Props
<PropsTable propMetaData={Checkbox.__docgenInfo} />

## a11y

Keyboard interactions: Space should activate the checkbox.

