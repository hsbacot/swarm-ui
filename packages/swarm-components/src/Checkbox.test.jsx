import React from 'react';
import { shallow } from 'enzyme';
import { Checkbox } from './Checkbox';

describe('Checkbox', () => {
	const testCases = [
		['Default', <Checkbox key="default">Check me</Checkbox>],
		[
			'Checked',
			<Checkbox key="checked" checked>
				Uncheck me
			</Checkbox>,
		],
		[
			'Disabled unchecked',
			<Checkbox key="disabled" disabled>
				Can&apos;t check me
			</Checkbox>,
		],
		[
			'Disabled checked',
			<Checkbox key="disabled-checked" disabled checked>
				Can&apos;t uncheck me
			</Checkbox>,
		],
	];

	test.each(testCases)('Snapshot: %s', (description, element) => {
		expect(shallow(element)).toMatchSnapshot();
	});
});
