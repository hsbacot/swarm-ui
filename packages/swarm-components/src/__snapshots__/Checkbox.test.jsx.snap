// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Snapshot: Checked 1`] = `
<label
  data-swarm-checkbox="default"
>
  <span
    aria-checked={true}
    data-swarm-checkbox-field="checked"
    role="checkbox"
  >
    <Icon
      color="#ffffff"
      shape="check"
      size="xs"
    />
  </span>
  <input
    checked={true}
    readOnly={true}
    type="checkbox"
    value="undefined"
  />
  <span>
    Uncheck me
  </span>
</label>
`;

exports[`Checkbox Snapshot: Default 1`] = `
<label
  data-swarm-checkbox="default"
>
  <span
    data-swarm-checkbox-field="unchecked"
    role="checkbox"
  />
  <input
    readOnly={true}
    type="checkbox"
    value="undefined"
  />
  <span>
    Check me
  </span>
</label>
`;

exports[`Checkbox Snapshot: Disabled checked 1`] = `
<label
  data-swarm-checkbox="disabled"
>
  <span
    aria-checked={true}
    data-swarm-checkbox-field="checked"
    role="checkbox"
  >
    <Icon
      color="var(--color-gray-6)"
      shape="check"
      size="xs"
    />
  </span>
  <input
    checked={true}
    disabled={true}
    readOnly={true}
    type="checkbox"
    value="undefined"
  />
  <span>
    Can't uncheck me
  </span>
</label>
`;

exports[`Checkbox Snapshot: Disabled unchecked 1`] = `
<label
  data-swarm-checkbox="disabled"
>
  <span
    data-swarm-checkbox-field="unchecked"
    role="checkbox"
  />
  <input
    disabled={true}
    readOnly={true}
    type="checkbox"
    value="undefined"
  />
  <span>
    Can't check me
  </span>
</label>
`;
